client/src/nv_ingest_client/cli/util/processing.py

```
@@ -313,24 +539,246 @@ def create_job_specs_for_batch(files_batch: List[str], tasks: Dict, client: NvIn
+        A future object representing an asynchronous job. The result of this job will be
+        processed once it completes.
+
+    futures_dict : Dict[concurrent.futures.Future, str]
+        A dictionary mapping future objects to job IDs. The job ID associated with the
+        provided future is retrieved from this dictionary.
+
+    Returns
+    -------
+    Dict[str, Any]
+
+    Raises
+    ------
+    RuntimeError
+        If the job result is invalid, this exception is raised with a description of the failure.
+
+    Notes
+    -----
+    - The `future.result()` is assumed to return a tuple where the first element is the actual
+      result (as a dictionary), and the second element (if present) can be ignored.
+    - Annotations in the result (if any) are logged for debugging purposes.
+    - The `check_ingest_result` function (assumed to be defined elsewhere) is used to validate
+      the result. If the result is invalid, a `RuntimeError` is raised.
+    - The function handles saving the result data to the specified output directory using the
+      `save_response_data` function.
+
+    Examples
+    --------
+    Suppose we have a future object representing a job, a dictionary of futures to job IDs,
+    and a directory for saving results:
+
+    >>> future = concurrent.futures.Future()
+    >>> futures_dict = {future: "job_12345"}
+    >>> job_id_map = {"job_12345": {...}}
+    >>> output_directory = "/path/to/save"
+    >>> result, retry_job_ids = handle_future_result(future, futures_dict, job_id_map, output_directory)
+
+    In this example, the function processes the completed job and saves the result to the
+    specified directory. If the job fails, it raises a `RuntimeError` and returns a list of
+    retry job IDs.
+
+    See Also
+    --------
+    check_ingest_result : Function to validate the result of the job.
+    save_response_data : Function to save the result to a directory.
+    """
+
+    result, _ = future.result()[0]
```

It seems like [client.py:_submit_job(...)](https://github.com/NVIDIA/nv-ingest/blob/main/client/src/nv_ingest_client/client/client.py#L407) can possibly raise an exception here. Since that code is ran in the ThreadPoolExecutor the exception would present itself here at the `.result()` call. I realize the JobState would already be set to failed and we could just let the event bubble on up but should we also add a try/except here too just to make the intent more clear?

---

I'll add a try catch that logs the error and re-raises.